tests/browser_test.go

```diff
@@ -69,15 +69,22 @@ func TestBrowserNewContext(t *testing.T) {
 func TestTmpDirCleanup(t *testing.T) {
 	t.Parallel()
 
-	const tmpDirPath = "./"
+	const tmpDirPath = "./1/"
+	err := os.Mkdir(tmpDirPath, os.ModePerm)
+	require.NoError(t, err)
+
+	defer func() {
+		err = os.Remove(tmpDirPath)
```

Comment:
From [the Go doc](https://pkg.go.dev/os#Remove): `os.Remove` remove removes the named file or (empty) directory. If there is an error, it will be of type *PathError. Should we use `os.RemoveAll` here?

We're also overwriting an existing error variable. There is no problem at this stage. However, if someone were to add another deferred function that also manipulates the same `err` variable, the order in which the deferred functions are defined would influence the behavior. Should we use another variable?

One last thing. If there's an assertion failure (e.g., require.NoError) before the deferred function is called, the cleanup code might not execute. This could leave the temporary directory on the file system. WDYT?

Reply:
> From [the Go doc](https://pkg.go.dev/os#Remove): os.Remove remove removes the named file or (empty) directory. Should we use os.RemoveAll here?

Yeah that's a good idea incase the test fails and the sub dir hasn't been removed.

> We're also overwriting an existing error variable. There is no problem at this stage. However, if someone were to add another deferred function that also manipulates the same err variable, the order in which the deferred functions are defined would influence the behavior. Should we use another variable?

👍 

> One last thing. If there's an assertion failure (e.g., require.NoError) before the deferred function is called, the cleanup code might not execute. This could leave the temporary directory on the file system. WDYT?

If there's an error on `MkdirTemp` then I would expect there to be no dir, but as a precaution i've moved the defer straight after calling `MkdirTemp`. Would it be better to work with `t.Cleanup` instead of defer though?

Resolved in: fd5d7e528d938939bd92f6f0322bf17c337586c4

Reply:
👍 Yes, `Cleanup` might make more sense since we might be queuing other `Cleanup` in our helpers like `testBrowser`.

Reply:
Resolved in: 1d4172704f7b1b0324bfccf4ec084c55e1296172